core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -77,21 +81,23 @@ public class SharePartitionManager implements AutoCloseable {
     private final Timer timer;
     private final int maxInFlightMessages;
     private final int maxDeliveryCount;
+    private final String shareGroupPersisterClassName;
 
     public SharePartitionManager(
             ReplicaManager replicaManager,
             Time time,
             ShareSessionCache cache,
             int recordLockDurationMs,
             int maxDeliveryCount,
-            int maxInFlightMessages
+            int maxInFlightMessages,
+            String shareGroupPersisterClassName
```

Comment:
This is getting to the point where a config class should be used. I'm happy for a parameter to be added in this case, but no more please.

Reply:
yeah, I agree. At a later stage, we would be removing this config altogether as well.